src/winmenu.c

```diff
@@ -71,14 +71,19 @@ Menu *CreateWindowMenu(ClientNode *np)
       AddWindowMenuItem(menu, _("Minimize"), MA_MINIMIZE, np, 0);
    }
 
-   if(!(np->state.status & STAT_FULLSCREEN)) {
+   if(!(np->state.status & STAT_FULLSCREEN) && 
+      !(np->state.status & STAT_MINIMIZED)  && 
+       (np->state.border & BORDER_MIN))
+   {
       if(np->state.status & STAT_SHADED) {
          AddWindowMenuItem(menu, _("Unshade"), MA_SHADE, np, 0);
       } else if(np->state.border & BORDER_SHADE) {
          AddWindowMenuItem(menu, _("Shade"), MA_SHADE, np, 0);
       }
       if((np->state.border & BORDER_MAX) &&
-         (np->state.status & (STAT_MAPPED | STAT_SHADED))) {
+         (np->state.status & STAT_MAPPED) && 
+        !(np->state.status & STAT_SHADED))
+      {
```

Comment:
I move shaded windows all the time (though I don't resize them or maximize them so much).  I don't know that there's any reason to remove the functionality though, which seems to work ok in my opinion.  I'd rather leave the option there since it doesn't hurt anything.